### PR TITLE
feat: redesign app

### PR DESCRIPTION
**Pull request by schmanu**
Update safe-react-components and use the new theme to redesign the app.
This requires an update of MUI to v5 and removes styled-components.
All wrapped MUI components were removed meaning that we have to use the native MUI components across the app.
    
Further Improvements:
- Uses dark mode if OS of user is set to dark appearance
- Improves design / flow of the app
- fixes an issue with multiple nfts from the safe collection all showing the same image.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,14 +17,20 @@
...
+    "@gnosis.pm/safe-react-gateway-sdk": "^3.5.2",
+    "@mui/icons-material": "^5.8.4",
+    "@mui/material": "^5.11.2",
+    "@mui/styles": "^5.11.9",
     "@openzeppelin/contracts": "^4.8.1",
     "@reduxjs/toolkit": "^1.8.1",
+    "@safe-global/safe-apps-provider": "^0.16.0",
+    "@safe-global/safe-apps-react-sdk": "^4.6.4",
+    "@safe-global/safe-apps-sdk": "^7.10.0",
+    "@safe-global/safe-react-components": "2.0.3",
```
Comment (bh2smith) [inline package.json:33]: Nice!

File: src/AppWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { Provider as ReduxProvider } from "react-redux";
+
+import App from "./App";
+import { useDarkMode } from "./hooks/useDarkMode";
+import errorIcon from "./static/error-icon.svg";
+import { store } from "./stores/store";
+
+export const AppWrapper = () => {
+  const isDarkMode = useDarkMode();
+  const themeMode = isDarkMode ? "dark" : "light";
```
Comment (bh2smith) [inline src/AppWrapper.tsx:13]: 🖤

File: src/AppWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+              },
+              ".ace_tooltip.ace_error": {
+                backgroundImage: "none !important",
+              },
+            }}
+          />
+          <SafeProvider
+            loader={
+              <>
+                <Typography>Waiting for Safe...</Typography>
```
Comment (bh2smith) [inline src/AppWrapper.tsx:39]: I am surprised that we need to put this ourselves. Seems like it should just be the default.

File: src/components/CSVForm.tsx [hunk 4]
```diff
@@ -1,33 +1,32 @@
...
-  gap: 8px;
-`;
 export interface CSVFormProps {}
 
 export const CSVForm = (props: CSVFormProps): JSX.Element => {
   return (
-    <Form>
-      <Text size="xl">
-        Send arbitrarily many distinct tokens, to arbitrarily many distinct accounts with various different values from
-        a CSV file in a single transaction.
```
Comment (bh2smith) [inline src/components/CSVForm.tsx:22]: Does this mean the app no longer had a short description in the App Store tile?

File: src/components/DonateDialog.tsx [hunk 5]
```diff
@@ -22,6 +39,7 @@ export const DonateDialog = ({
   const { safe } = useSafeAppsSDK();
   const dispatch = useAppDispatch();
   const csvContent = useCsvContent();
+  const darkMode = useDarkMode();
```
Comment (bh2smith) [inline src/components/DonateDialog.tsx:42]: 🖤🤘🏾😈🏴‍☠️

File: src/components/Summary.tsx [hunk 6]
```diff
@@ -14,10 +20,22 @@ export const Summary = (props: SummaryProps): JSX.Element => {
   const { assetTransfers, collectibleTransfers } = props;
   const assetTxCount = assetTransfers.length;
   const collectibleTxCount = collectibleTransfers.length;
+  const darkMode = useDarkMode();
```
Comment (bh2smith) [inline src/components/Summary.tsx:23]: 🖤💯👌

**Review by bh2smith — APPROVED** [review]
Looks like mostly imports dependencies and style boxes. No need to hold this back! Seems we lost the app tile text and have introduced dark mode (a key ingredient). Love it.

File: src/components/CSVForm.tsx (hunk 4 above)
Comment (schmanu) [reply src/components/CSVForm.tsx:22]: No the description in the AppStore is in our manifest file

**Pull request merged**